obspy/imaging/maps.py

```diff
@@ -160,134 +163,142 @@ def plot_basemap(lons, lats, size, color, labels=None, projection='global',
+                if round_pos < 0:
+                    factor = 10 ** (abs(round_pos))
+                    delta = round(2. * dval / N / factor) * factor / 2
+                else:
+                    delta = round(2. * dval / N, round_pos) / 2
+                new_val1 = np.ceil(val1 / delta) * delta
+                new_val2 = np.floor(val2 / delta) * delta
+                N = (new_val2 - new_val1) / delta + 1
+                return np.linspace(new_val1, new_val2, N)
+
+            N1 = int(np.ceil(height / max(width, height) * 8))
+            N2 = int(np.ceil(width / max(width, height) * 8))
+            parallels = linspace2(lat_0 - height / 2 / deg2m_lat,
+                                  lat_0 + height / 2 / deg2m_lat, N1)
+
+            # Old basemap versions have problems with non-integer parallels.
+            try:
+                bmap.drawparallels(parallels, labels=[0, 1, 1, 0])
+            except KeyError:
+                parallels = sorted(list(set(map(int, parallels))))
+                bmap.drawparallels(parallels, labels=[0, 1, 1, 0])
+
+            if min(lons) < -150 and max(lons) > 150:
+                lon_0 %= 360
+            meridians = linspace2(lon_0 - width / 2 / deg2m_lon,
+                                  lon_0 + width / 2 / deg2m_lon, N2)
+            meridians[meridians > 180] -= 360
+            bmap.drawmeridians(meridians, labels=[1, 0, 0, 1])
+        else:
+            msg = "Projection '%s' not supported." % projection
+            raise ValueError(msg)
+
+        # draw coast lines, country boundaries, fill continents.
+        map_ax.set_axis_bgcolor(water_fill_color)
+        bmap.drawcoastlines(color="0.4")
+        bmap.drawcountries(color="0.75")
+        bmap.fillcontinents(color=continent_fill_color,
+                            lake_color=water_fill_color)
+        # draw the edge of the bmap projection region (the projection limb)
+        bmap.drawmapboundary(fill_color=water_fill_color)
+        # draw lat/lon grid lines every 30 degrees.
+        bmap.drawmeridians(np.arange(-180, 180, 30))
+        bmap.drawparallels(np.arange(-90, 90, 30))
+        fig.bmap = bmap
     else:
-        msg = "Projection '%s' not supported." % projection
-        raise ValueError(msg)
-
-    # draw coast lines, country boundaries, fill continents.
-    map_ax.set_axis_bgcolor(water_fill_color)
-    bmap.drawcoastlines(color="0.4")
-    bmap.drawcountries(color="0.75")
-    bmap.fillcontinents(color=continent_fill_color,
-                        lake_color=water_fill_color)
-    # draw the edge of the bmap projection region (the projection limb)
-    bmap.drawmapboundary(fill_color=water_fill_color)
-    # draw lat/lon grid lines every 30 degrees.
-    bmap.drawmeridians(np.arange(-180, 180, 30))
-    bmap.drawparallels(np.arange(-90, 90, 30))
+        map_ax = fig.axes[0]
+        fig = map_ax.get_figure()
+        bmap = fig.bmap
```

Comment:
Should the case that someone uses a completely new `Figure` without bmap be handled as well?